File: app/src/main/java/com/htoyama/likit/AppComponent.java
package com.htoyama.likit;

import com.htoyama.likit.data.DataModule;
import com.htoyama.likit.domain.liked.LikedRepository;
import com.htoyama.likit.domain.tag.TagRepository;
import com.htoyama.likit.domain.user.UserRepository;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.support.AndroidSupportInjectionModule;

/**
 * Created by htoyama on 2017/04/23.
 */

@Singleton
@Component(modules ={
    AndroidSupportInjectionModule.class,
    AppModule.class,
    BuildersModule.class })
public interface AppComponent {

  @Component.Builder interface Builder {
    @BindsInstance
    Builder application(App application);
    AppComponent build();
  }

  void inject(App app);

  /*
  LikedRepository likedRepository();
  TagRepository tagRepository();
  UserRepository userRepository();
  */
}

File: app/src/main/java/com/htoyama/likit/AppComponent.kt
package com.htoyama.likit

import com.htoyama.likit.data.net.FavoriteService
import com.htoyama.likit.data.prefs.AppSetting
import com.htoyama.likit.data.sqlite.likedtweet.LikedTweetTableGateway
import com.htoyama.likit.domain.likedtweet.LikedTweetRepository
import com.htoyama.likit.domain.tag.TagRepository
import com.htoyama.likit.domain.user.UserRepository
import com.htoyama.likit.ui.auth.AuthActivity
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AppModule::class))
interface  AppComponent {
  fun inject(activity: AuthActivity)
  fun inject(activity: MainActivity)

  fun tweetTableGateway(): LikedTweetTableGateway
  fun appSetting(): AppSetting
  fun favoriteService(): FavoriteService
  fun likedRepository(): LikedTweetRepository
  fun tagRepository(): TagRepository
  fun userRepository(): UserRepository
}
